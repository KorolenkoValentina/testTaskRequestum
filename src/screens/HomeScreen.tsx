
import React, { useEffect, useState } from 'react';
import axios from 'axios';

import { Text } from 'react-native';
import { clearSessionData} from '../utils/storage'
import { UserData, NavigationRoutes } from '../components/interfaces';

import styled from 'styled-components/native';

import LetterR from '../image/letterR.svg'


const HomeScreen: React.FC<{ navigation: any }> = ({ navigation }) => {
  const [userData, setUserData] = useState<UserData | null>(null);

  useEffect(() => {
    const fetchData = async () => {
      try {
        // const userData = await getUserData();
        // if (!userData) {
          
        //   navigation.navigate(NavigationRoutes.SIGN_UP);
        //   return;
        // }
        
        const response = await axios.get<UserData>('https://jsonplaceholder.typicode.com/users/1');
        setUserData(response.data);
      } catch (error) {
        console.error('Data retrieval error:', error);
      }
    };
  
    fetchData();
  }, []);
 
  const handleLogOut = async () => {
    try {
      await clearSessionData()
   
      navigation.navigate(NavigationRoutes.SPLASH);
    } catch (error) {
      console.error('Logout error:', error);
    }
  };
    
  
    return (
      
      <Container>
        <BgImage source={require('../image/bg.png')} />

        <LetterR/>
        {userData ? (
        <ContentContainer>
          <Subtitle>You’re loggen in now </Subtitle>
          <Header>{userData.name}  username:{userData.username}</Header>
          <Subtitle>Email: {userData.email}</Subtitle>
          <Subtitle>Phone: {userData.phone}</Subtitle>

        </ContentContainer>
         ) : (
          <Text>Завантаження...</Text>
        )}
        <LogOutButton onPress={handleLogOut}>
          <ButtonText>Log Out</ButtonText>
        </LogOutButton>
       

             
      </Container>
      
    );
  };

export default HomeScreen;



const Container = styled.View`
  flex: 1;
  justify-content: space-evenly;
  align-items: center;
  
  
`;
const BgImage = styled.Image`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const ContentContainer = styled.View`
  position: absolute;
  top: 50%;
  

`;

const Header = styled.Text`
  font-size: 18px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.white};
  margin-bottom: 20px;
  text-align: center;
`;

const Subtitle = styled.Text`
  font-size: 11px;
  color: ${({ theme }) => theme.colors.lightGray};
  margin-bottom: 24px;
  text-align: center;
`;

const LogOutButton = styled.TouchableOpacity`
  width: 80%;
  background-color: ${({ theme }) => theme.colors.pink};
  border-radius: 5px;
  padding: 15px;
  margin-bottom: 50px;
  shadow-color: ${({ theme }) => theme.colors.pink}; 
  shadow-offset: 0px 12px; 
  shadow-opacity: 0.58; 
  shadow-radius: 16px; 
  elevation: 5; /* Elevation для Android */;
`;
const ButtonText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  text-align: center;
  font-size: 12px; 
`;
