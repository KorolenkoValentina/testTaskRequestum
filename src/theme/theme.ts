const theme = {
    colors: {
        white : '#FFFFFF',
        lightGray: '#818181',
        pink: '#EB0057',
        dark: '#262727',
    },
    
};
  
export default theme;