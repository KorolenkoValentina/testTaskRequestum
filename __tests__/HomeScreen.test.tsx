import React from 'react';
import axios from 'axios';
import renderer from 'react-test-renderer';
import { act } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import theme from '../src/theme/theme';
import HomeScreen from '../src/screens/HomeScreen';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
  jest.mock('axios', () => ({
get: jest.fn(),
}));



describe('HomeScreen component', () => {
    const mockNavigation = {
      navigate: jest.fn(),
    };
  
    test('matches snapshot', () => {
      const tree = renderer.create(
        <ThemeProvider theme={theme}>
          <HomeScreen navigation={mockNavigation} />
        </ThemeProvider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('navigates to Splash screen on log out', async () => {
      const component = renderer.create(
        <ThemeProvider theme={theme}>
          <HomeScreen navigation={mockNavigation} />
        </ThemeProvider>
      );
      const instance = component.root;
  
      const logoutButton = instance.findByProps({ testID: 'logoutButton' });
      await logoutButton.props.onPress(); 
  
      expect(mockNavigation.navigate).toHaveBeenCalledWith('Splash');
  });

  test('fetches user data correctly', async () => {
    const mockedAxiosGet = axios.get as jest.MockedFunction<typeof axios.get>;
    const mockUserData = {
      name: 'John Doe',
      username: 'johndoe',
      email: 'john@example.com',
      phone: '123-456-7890',
    };
  
    mockedAxiosGet.mockResolvedValue({ data: mockUserData });
  
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <HomeScreen navigation={mockNavigation} />
      </ThemeProvider>
    );
  
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0)); 
    });
  
    expect(axios.get).toHaveBeenCalledWith('https://jsonplaceholder.typicode.com/users/1');
    expect(component.toJSON()).toMatchSnapshot();
  });


   
})