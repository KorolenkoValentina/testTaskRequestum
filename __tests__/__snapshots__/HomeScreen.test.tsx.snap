// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen component fetches user data correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-evenly",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  />
  <SvgMock />
  <View
    style={
      {
        "position": "absolute",
        "top": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#818181",
          "fontSize": 11,
          "marginBottom": 24,
          "textAlign": "center",
        }
      }
    >
      You’re loggen in now 
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      John Doe
        username:
      johndoe
    </Text>
    <Text
      style={
        {
          "color": "#818181",
          "fontSize": 11,
          "marginBottom": 24,
          "textAlign": "center",
        }
      }
    >
      Email: 
      john@example.com
    </Text>
    <Text
      style={
        {
          "color": "#818181",
          "fontSize": 11,
          "marginBottom": 24,
          "textAlign": "center",
        }
      }
    >
      Phone: 
      123-456-7890
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#EB0057",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "elevation": 5,
        "marginBottom": 50,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "shadowColor": "#EB0057",
        "shadowOffset": {
          "height": 12,
          "width": 0,
        },
        "shadowOpacity": 0.58,
        "shadowRadius": 16,
        "width": "80%",
      }
    }
    testID="logoutButton"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      Log Out
    </Text>
  </View>
</View>
`;

exports[`HomeScreen component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-evenly",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  />
  <SvgMock />
  <Text>
    Завантаження...
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#EB0057",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "elevation": 5,
        "marginBottom": 50,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "shadowColor": "#EB0057",
        "shadowOffset": {
          "height": 12,
          "width": 0,
        },
        "shadowOpacity": 0.58,
        "shadowRadius": 16,
        "width": "80%",
      }
    }
    testID="logoutButton"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      Log Out
    </Text>
  </View>
</View>
`;
